x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -24,15 +24,77 @@ import {
   AllReportersFindRequest,
   CasesByAlertIDRequest,
   CasesByAlertIDRequestRt,
+  AlertResponse,
+  AttributesTypeAlerts,
 } from '../../../common/api';
-import { countAlertsForID, flattenCaseSavedObject } from '../../common';
+import { countAlertsForID, flattenCaseSavedObject, getIDsAndIndicesAsArrays } from '../../common';
 import { createCaseError } from '../../common/error';
 import { ENABLE_CASE_CONNECTOR } from '../../../common/constants';
-import { CasesClientArgs } from '..';
+import { CasesClient, CasesClientArgs } from '..';
 import { Operations } from '../../authorization';
 import { combineAuthorizedAndOwnerFilter } from '../utils';
 import { CasesService } from '../../services';
 
+const normalizeAlertResponse = (alerts: Array<SavedObject<AttributesTypeAlerts>>): AlertResponse =>
+  alerts.reduce((acc: AlertResponse, alert) => {
+    const { ids, indices } = getIDsAndIndicesAsArrays(alert.attributes);
+
+    if (ids.length !== indices.length) {
+      return acc;
+    }
+
+    return [
+      ...acc,
+      ...ids.map((id, index) => ({
+        id,
+        index: indices[index],
+        attached_at: alert.attributes.created_at,
+      })),
+    ];
+  }, []);
+
+export interface GetAllAlertsAttachToCase {
+  caseId: string;
+}
+
+export const getAllAlertsAttachToCase = async (
+  { caseId }: GetAllAlertsAttachToCase,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+): Promise<AlertResponse> => {
+  const { unsecuredSavedObjectsClient, authorization, caseService } = clientArgs;
+  const theCase = await casesClient.cases.get({
```

Comment:
I don't think we need to request the case do we?

If we do, could we just use the cases service directly instead of doing it through the client? Otherwise we'll be performing another `ensureAuthorized` call.

Reply:
You are right about that. I wanted to ask you about it. One advantage I can think of is that it will through an 404 if the case does not exists which is more align with the REST paradigm. @legrego Is there any security consideration with it?

Reply:
Thanks @legrego we'll keep the `casesClient.cases.get()` call in as this will do an additional authorization check to ensure that the user has access to the parent case as well as the alerts that we return.